### PR TITLE
sessionctx/variable: improve test coverage

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/35051

Problem Summary:

This is a function neutral change that moves around some tests (and a helper function). It improves the coverage of sessionctx/variable from 42.8% to 45.5% - but importantly the testing of the sysvar framework (variable.go) improves from 78.1% to 91.9%. Most of the lack of coverage is on sysvar.go - which is still at 31.4%. This is mostly because this code is indirectly tested in the other relevant packages.

### What is changed and how it works?

Test improvements.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -1070,13 +1070,3 @@ var (
...
-
-// switchDDL turns on/off DDL in an instance.
-func switchDDL(on bool) error {
-	if on && EnableDDL != nil {
-		return EnableDDL()
-	} else if !on && DisableDDL != nil {
-		return DisableDDL()
-	}
-	return nil
-}
```
Comment (morgo) [inline sessionctx/variable/tidb_vars.go:1082]: This is moved to varsutil.go, with other helpers.

**Review by CbcWestwolf — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by hawkingrei** [comment]
@pingcap/tidb-configuration-reviewer PTAL

**Comment by hawkingrei** [comment]
/hold 

CI has some problem.

**Comment by hawkingrei** [comment]
@pingcap/tidb-configuration-reviewer  PTAL

**Comment by hawkingrei** [comment]
@bb7133 @qw4990 @XuHuaiyu PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**